pkg/securityhubcollector/security_hub_collector.go

```diff
@@ -53,39 +53,82 @@ func (h *HubCollector) GetSecurityHubFindings() ([]*securityhub.AwsSecurityFindi
 	return outputList, nil
 }
 
-// ConvertFindingsToCSV - converts select Finding fields to CSV
-func (h *HubCollector) ConvertFindingsToCSV(findings []*securityhub.AwsSecurityFinding) error {
+// ConvertFindingToRows - converts a single finding to the record format we're using
+func (h *HubCollector) ConvertFindingToRows(finding *securityhub.AwsSecurityFinding) [][]string {
+	var output [][]string
+
+	// Each finding may have multiple resources, so we need to iterate through
+	// them and pull the relevant bits; we will create two lines for a single
+	// finding that has two resources, with only the resource different.
+	for _, r := range finding.Resources {
+		// Here we compile a single record, which is a representation of
+		// the row we want to output into the CSV for this resource in
+		// the finding.
+		var record []string
+		record = append(record, "Team TBD")
+		record = append(record, *r.Type)
+		record = append(record, *finding.Title)
+		record = append(record, *finding.Description)
+		record = append(record, *finding.Severity.Label)
+		record = append(record, *finding.Remediation.Recommendation.Text)
+		record = append(record, *finding.Remediation.Recommendation.Url)
+		record = append(record, *r.Id)
+		record = append(record, *finding.AwsAccountId)
+		record = append(record, *finding.Compliance.Status)
+		record = append(record, *finding.RecordState)
+
+		// Each record *may* have multiple findings, so we make a list of
+		// records and that's what we'll output.
+		output = append(output, record)
+	}
+
+	return output
+
+}
+
+// WriteFindingsToOutput - takes a list of security
+func (h *HubCollector) WriteFindingsToOutput(findings []*securityhub.AwsSecurityFinding) (err error) {
 	// Try to create the output file we got from the collector object
 	f, err := os.Create(h.Outfile)
 	if err != nil {
-		return err
+		return
```

Comment:
Curious why you choose not to return `err`?

Reply:
It does return error automatically because err is a named output -- I took this approach based on the suggestions here: https://www.joeshaw.org/dont-defer-close-on-writable-files/